acme.sh

```diff
@@ -1003,7 +1003,7 @@ _sign() {
 
   _sign_openssl="${ACME_OPENSSL_BIN:-openssl} dgst -sign $keyfile "
 
-  if grep "BEGIN RSA PRIVATE KEY" "$keyfile" >/dev/null 2>&1; then
+  if egrep -o "BEGIN( RSA)? PRIVATE KEY" "$keyfile" >/dev/null 2>&1; then
```

Comment:
why is this change necessary?

Reply:
Trying to not repeat code I saw that there already was `_sign` function. The private key as being used by TransIP doesn't start with "BEGIN RSA PRIVATE KEY" but rather with "BEGIN PRIVATE KEY" (which is also valid), therefore - instead of incorporating a duplicate of the sign command I propose to also allow the `_sign` command to work for this type of key (and as it's only an extension it doesn't break anything)

Reply:
don't use `egrep`, it's not portable.

please use 2  'if' conditions. like:


```
if grep "BEGIN RSA PRIVATE KEY" "$keyfile" >/dev/null 2>&1; || if grep "BEGIN RSA PRIVATE KEY" "$keyfile" >/dev/null 2>&1;
```

Reply:
Happy to fix this. But there's also a _egrep_o() function in acme.sh on line 878 which uses egrep (which issued at various other places). I'm not tied to egrep in anyway so I'll make the changes if that works best.

Reply:
`_egrep_o` is just for searching simple regex strings.  It's not reliable or robust enough.

Reply:
Fixed